### ISSUE TITLE
Feature request : pwnlib.asm.disasm multiple code segments

### ISSUE THREAD

**Issue by h4ck3rm1k3** [Gallopsled/pwntools#790]
It would be more time efficient to be able to disasm multiple code strings by putting them into different segments and invoking the binutils only once.

**Comment by h4ck3rm1k3** [comment]
Here is how I did it

```
.test1:
.long 0x000000
.test2:
.long 0x100000
.test3:
.long 0x010000
.test4:
.long 0x110000

```

then used as to assemble that and objdump to get the objects with mutiple segments

you can also make the objects longer 

```
.test1:
.long 0x000000
.long 0x100000
.long 0x010000
.test4:
.long 0x110000

```

**Comment by zachriggle** [comment]
I think that if you need a time-efficient disassembler, you should look at
Binary Ninja's Python API (or use Capstone or IDA).

Pwntool's disassembler is convenience-efficient, not performance-efficient.

Bulk disassembly isn't something pwntools is well-situated to do.

**Comment by h4ck3rm1k3** [comment]
I have a solution that does not need pwntools at all, as described. It should be possible to do with pwntools as well.

**Comment by zachriggle** [comment]
Sure, if you've got a Pull Request for this, we can work with you to get the feature accepted.

**Comment by zachriggle** [comment]
Closing this, since it's not a priority for me.  I can reopen this if a Pull Request is submitted.
